darshan-util/pydarshan/darshan/backend/cffi_backend.py

```diff
@@ -768,15 +777,17 @@ def log_get_derived_metrics(rec_dict, mod_name, nprocs):
                            "to retrieve additional information from the stderr "
                            "stream.")
     derived_metrics = ffi.new("struct darshan_derived_metrics *")
-    total_record = ffi.new(_structdefs[mod_name].replace("**", "*"))
+    summary_rbuf = ffi.new(_structdefs[mod_name].replace("**", "*"))
     r = libdutil.darshan_accumulator_emit(darshan_accumulator[0],
                                           derived_metrics,
-                                          total_record)
+                                          summary_rbuf)
     libdutil.darshan_accumulator_destroy(darshan_accumulator[0])
     if r != 0:
         raise RuntimeError("A nonzero exit code was received from "
                            "darshan_accumulator_emit() at the C level. "
                            "It may be possible "
                            "to retrieve additional information from the stderr "
                            "stream.")
-    return derived_metrics
+
+    summary_rec = _make_generic_record(summary_rbuf, mod_name, dtype)
+    return derived_metrics, summary_rec
```

Comment:
Finally, for the comment:

> Can we convert the derived metrics structure returned by accumulate_records() into a more Pythonic object? summary_record is a dictionary now (i.e., not a cdata object), so seems that it would be cleaner if both return values are typical Python things, rather than derived_metrics being a C-like structure.

Maybe that should be a separate issue as well--if it substantially reduces lines of code maybe. It is true that I don't like how there is a `struct` than then contains an `array` which has to be manually indexed based on knowledge of the C code to pull out some counter values. Maybe sketch out a design in another issue if you plan to flatten to a custom `class` or maybe even `dataclass` is appropriate... though dunno where it sits on priority side.

Reply:
A summary record is orthogonal to derived metrics, though they are both notionally based on aggregating sets of module records.

_derived metrics_: Calculates a set of statistics describing high-level I/O characteristics of a set of records, including things like total bytes read/written, total time spent doing I/O, bandwidth calculations, etc. Darshan can provide derived metrics for all of its traditional modules, even though they have different counter names, etc. It's just a way to generalize I/O characteristics of different modules into general I/O behavior that is no longer module specific, if that makes sense.

_summary record_: This literally generates an entirely new module record that is the aggregation of a set of given module records. So, it's not calculating derived metrics, it's just creating a new POSIX record that is the aggregation of a set of input POSIX records, for instance. You could use something like this to turn a bunch of per-process records referencing a file into a single aggregate record for that file (giving you total operation counts, access sizes, etc.).

As an aside, I think basically all of the plots in our existing job summary report could simply be updated to use an  summary record for input, which would simplify things a lot, I think (completely ignoring the issues related to whether this aggregation should be in Python or C -- the aggregation is useful and is now hidden behind an API, we could always move it in the future but I don't think it's really relevant here).

Reply:
> Maybe that should be a separate issue as well--if it substantially reduces lines of code maybe. It is true that I don't like how there is a struct than then contains an array which has to be manually indexed based on knowledge of the C code to pull out some counter values. Maybe sketch out a design in another issue if you plan to flatten to a custom class or maybe even dataclass is appropriate... though dunno where it sits on priority side.

New issue in gh-924